### DISCUSSION TITLE
The range `start` must be before or equal to the range `end` when using @Timestamp filter

### DISCUSSION THREAD

**Discussion by SamLJG** [datalust/seq-tickets#1239] [Help]
Using Seq largely through the Seq API via .NET, we have encountered an issue where we are getting issues with setting from and to date parameters.

To try and keep queries responsive, we apply a 'failsafe' event age limit of '@Timestamp > Now() - 30d' as typically we're only interested in fairly recent events but do keep events for quite a lot longer than this. We've noticed however that if we combine a timestamp based filter and a from/to limit on the /api/events/signal endpoint, we get a 400 response with the following content:

```
The range `start` must be before or equal to the range `end`.
```

This seems to happen if the from/to and timestamp filters don't overlap. While this shouldn't return any results, I wouldn't expect it to return a 400 response either. Ultimately, if they don't overlap the query is nonsensical (get results from over a month ago where also less than 30 days old).

Our solution is to simply not apply the @Timestamp filter when restricting results using the from/to arguments. Fortunately, leaving it in when a date filter was set was an oversight in the first place. Nonetheless, this did cause a few headaches and didn't feel like expected behaviour.

A sample request that seems to re-create the issue on a fresh Seq instance (2021.2.5647)

```
/#/events?from=2021-05-02T23:00:00.000Z&to=2021-05-06T23:00:00.000Z&filter=@Timestamp%20%3E%20Now()-30d
```

**Comment by nblumhardt** [comment]
Hi Sam,

Thanks for the heads-up; I agree that this is awkward and a little unexpected.

I'll have a dig into it and see how we might be able to improve the experience, here!

Best regards,
Nick
